test/CheckedC/dynamic-checks/struct-increment-code-gen.c

```diff
@@ -109,7 +115,11 @@ void f1(void) {
   // CHECK-AST-NEXT: IntegerLiteral {{.*}} 2
 
   // CHECK-IR: [[REG1:%[a-zA-Z0-9.]*]] = load %struct.S1*, %struct.S1** @gp3
+  // NOTE: no check here, because we don't yet do checks at pointer arithmetic
```

Comment:
Ok, I've had a look. They should go here: https://github.com/Microsoft/checkedc-clang/blob/master/lib/CodeGen/CGExprScalar.cpp#L2454-L2456

The reason this isn't trivial is that it turns out this is hit when generating the upper and lower bounds for the bounds check, which we don't want. A full fix is a different PR (which would prevent emitting non-null checks within a bounds check, as at any time we emit the bounds check, we have done a non-null check of the base ptr just before).